pkg/apis/core/v1/helper/qos/qos.go

```diff
@@ -90,85 +91,86 @@ func getQOSResources(list v1.ResourceList) sets.Set[string] {
 // TODO(ndixita): Refactor ComputePodQOS into smaller functions to make it more
 // readable and maintainable.
 func ComputePodQOS(pod *v1.Pod) v1.PodQOSClass {
+	requests, limits, isGuaranteed := collectPodResources(pod)
+	if len(requests) == 0 && len(limits) == 0 {
+		return v1.PodQOSBestEffort
+	}
+	if isGuaranteed && apiequality.Semantic.DeepEqual(requests, limits) {
```

Comment:
I see, so `apiequality.Semantic.DeepEqual()` performs the same comparison as the previous implementation.
https://github.com/kubernetes/kubernetes/blob/6f37102668f58439f712d0ccfcc88cf896097660/staging/src/k8s.io/apimachinery/pkg/api/equality/semantic.go#L30-L35